### PR TITLE
Download Diagnosis Keys from GAEN Server

### PR DESCRIPTION
**Pull request by mattThousand**
### Why
We'd like the iOS client to download diagnosis keys from the GAEN server in order to submit them to the `Exposure Notifications` library on device and determine whether the user has been exposed to COVID-19.

### This Commit
This commit updates the `ExposureManager` class to request an index file from the GAEN server containing a list of URLs for a given set of regions, and after parsing that file, download each diagnosis key.

### CODE REVIEW

File: ios/BT/API/APIClient.swift [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
 
-  let postUrl: URL
-  let pullUrl: URL
+  let postKeysUrl: URL
+  let downloadKeyFile: URL
+  let indexFileUrl: URL
   static let shared = APIClient(
-    postUrl: URL(string: "https://exposure-2kabcv6c4a-uc.a.run.app")!,
-    pullUrl: URL(string: "https://federationout-2kabcv6c4a-uc.a.run.app")!
+    postKeysUrl: URL(string: "https://exposure-2kabcv6c4a-uc.a.run.app")!,
```
Comment (johnschoeman) [inline ios/BT/API/APIClient.swift]: would these strings make sense in an env file or something?

File: ios/BT/API/APIClient.swift [hunk 2]
```diff
@@ -81,14 +108,35 @@ private extension APIClient {
     static let errorMessage = "error_description"
   }
 
+  func downloadRequest<T: APIRequest>(for request: T, fileName: String) -> DownloadRequest {
```
Comment (johnschoeman) [inline ios/BT/API/APIClient.swift:111]: Not sure if this is a swift thing, but should there be 2 `downloadRequest` functions?

File: ios/Podfile.lock [hunk 3]
```diff
@@ -199,7 +199,7 @@ PODS:
     - React
   - react-native-geolocation (2.0.2):
     - React
-  - react-native-safe-area-context (3.0.5):
```
Comment (johnschoeman) [inline ios/Podfile.lock]: not sure this change is necessary to commit.

**Review by johnschoeman — APPROVED** [review]
LGTM! This is a crazy amount of work!

File: ios/Podfile.lock (hunk 3 above)
Comment (mattThousand) [reply ios/Podfile.lock]: ah - good catch, this one snuck in unintentionally

File: ios/BT/API/APIClient.swift (hunk 2 above)
Comment (mattThousand) [reply ios/BT/API/APIClient.swift:111]: good question - yes, it's idiomatic (depending on who you ask) to have 2 functions with identical names but different signatures. The compiler (autocomplete) helps you out during development to ensure that you're calling the intended one

File: ios/BT/API/APIClient.swift (hunk 1 above)
Comment (mattThousand) [reply ios/BT/API/APIClient.swift]: It would indeed. Will make a new commit.

File: ios/BT/API/APIClient.swift (hunk 2 above)
Comment (johnschoeman) [reply ios/BT/API/APIClient.swift:111]: pretty neat! elixir has a similar feature and its awesome!

**Pull request merged**